### PR TITLE
Our perf thresholds have changed

### PR DESCRIPTION
**Pull request by Lukasa**

### CODE REVIEW

**Review by dnadoba — APPROVED** [review]

**Comment by Lukasa** [comment]
The retain/release counts in sync code have become very unstable, and I don't think they're providing us much value in comparison to the time I'm spending trying to dial them in. So I propose we just remove them.

**Comment by dnadoba** [comment]
Fair! If they are not stable, even in sync code, we need to disabled them. It is surprising that this now effects sync code. @hassila are there any known issues?
I'm only aware of https://github.com/ordo-one/package-benchmark/issues/189

File: Benchmarks/Thresholds/5.10/CertificatesBenchmark.Parse_WebPKI_Roots_from_DER.p90.json [hunk 1]
```diff
@@ -1,8 +1,8 @@
 {
-  "mallocCountTotal" : 5506,
+  "mallocCountTotal" : 4851,
   "memoryLeaked" : 0,
   "readSyscalls" : 0,
-  "releaseCount" : 120084,
-  "retainCount" : 108506,
+  "releaseCount" : 120271,
+  "retainCount" : 109425,
```
Comment (dnadoba) [inline Benchmarks/Thresholds/5.10/CertificatesBenchmark.Parse_WebPKI_Roots_from_DER.p90.json:6]: Can we remove the retain/release counts from the output as well please?

**Review by dnadoba — CHANGES_REQUESTED** [review]

File: Benchmarks/Thresholds/5.10/CertificatesBenchmark.Parse_WebPKI_Roots_from_DER.p90.json (hunk 1 above)
Comment (Lukasa) [reply Benchmarks/Thresholds/5.10/CertificatesBenchmark.Parse_WebPKI_Roots_from_DER.p90.json]: Done.

**Review by dnadoba — APPROVED** [review]

**Comment by hassila** [comment]
Nothing that I know of that should make them unstable (possibly if you run any sampling metric like thread counts etc? If you run with only ARC metrics enabled is it still unstable?). Then of course it doesn't always add up as mentioned in the related case (some objects are also created with initial ref count which is another factor why we sometime see more releases).

**Comment by Lukasa** [comment]
I don't think we had any sampling metrics enabled: we recorded syscalls, allocations, and retains/releases. And the issue wasn't an adding-up problem: it was that the values changed from run-to-run.

**Comment by hassila** [comment]
Yeah, that's very strange then, that they don't add up is known, but they should be stable.

**Comment by hassila** [comment]
(to be sure to isolate it, you might try running with only arc metrics enabled, then I can't really see how any other parts of the benchmark infrastructure could affect it much)

**Comment by dnadoba** [comment]
@Lukasa have you investigated why the thresholds have changed? How have you updated them?

I'm running `./dev/update-benchmark-thresholds` locally which produces a stable output that I have opened a PR with just now: https://github.com/apple/swift-certificates/pull/184
However, CI doesn't agree on these. These numbers seem to be stable on CI as well.
Docker is used locally and on CI so I wouldn't expect any difference.

**Comment by Lukasa** [comment]
My current theory is that because we aren't scaling the iterations, we're extremely sensitive to minor variations. We should have things settle down if we scale the iteration count by kilo or mega.

**Pull request merged**